drivers/frequency/ad9528/ad9528.c

```diff
@@ -687,20 +1160,36 @@ int32_t ad9528_setup(struct ad9528_dev **device,
 		ret = ad9528_spi_write_n(dev, AD9528_STAT_PIN_EN,
 					 stat_en_mask);
 		if (ret < 0)
-			return ret;
+			goto error;
 	}
 
 	ret = ad9528_io_update(dev);
 	if (ret < 0)
-		return ret;
+		goto error;
 
 	ret = ad9528_sync(dev);
 	if (ret < 0)
-		return ret;
+		goto error;
+
+	ret = jesd204_dev_register(&dev->jdev, &jesd204_ad9528_init);
+	if (ret < 0)
+		goto error;
+	priv = jesd204_dev_priv(dev->jdev);
+	priv->device = dev;
 
 	*device = dev;
 
 	return 0;
+
+error:
+
+	for (i = 0; i < AD9528_NUM_CHAN; i++) {
+		no_os_free(clocks[i]);
+		clocks[i] = NULL;
+	}
+	no_os_free(dev);
+	dev = NULL;
```

Comment:
This is not useful in this specific case (no longer used local variable).

Reply:
Removed all occurrences.